autogtp/Game.cpp

```diff
@@ -0,0 +1,211 @@
+/*
+    This file is part of Leela Zero.
+    Copyright (C) 2017 Marco Calignano
+
+    Leela Zero is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Leela Zero is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Leela Zero.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include<QUuid>
+#include "Game.h"
+
+Game::Game(QString weights, QTextStream *out) :
+    QProcess(),
+    output(out),
+    cmdLine("./leelaz"),
+    timeSettings("time_settings 0 1 0"),
+    readCount(0),
+    resignation(false),
+    blackToMove(true),
+    blackResigned(false),
+    passes(0),
+    moveNum(0)
+{
+#ifdef WIN32
+    cmdLine.append(".exe");
+#endif
+    cmdLine.append(" -g -q -n -m 30 -r 0 -w ");
+    cmdLine.append(weights);
+    cmdLine.append(" -p 400 --noponder");
+    QString randomName(QUuid::createUuid().toRfc4122().toHex());
+    sgfName = randomName + ".sgf";
```

Comment:
If you make randomName the member variable, then sgfName and trainingName don't need to be and can be constructed where they are used.